src/views/AddProductModal.vue

```diff
@@ -20,8 +20,8 @@
           </ion-thumbnail>
           <ion-label>
             <!-- Honouring the identifications set by the user on the settings page -->
-            <h2>{{ product[productIdentificationPref.primaryId] }}</h2>
-            <p>{{ product[productIdentificationPref.secondaryId] }}</p>
+            <h2>{{ productHelpers.getProductIdentificationValue(productIdentificationPref.primaryId, getProduct(product.productId)) }}</h2>
```

Comment:
Use this function from dxp-components package and update it on all the places.
Also add a check on all the places to display productName if primary identifier is missing.
https://github.com/hotwax/fulfillment-pwa/blob/445f1faccc9cb7435c37b2758f4e5d9637feb02a/src/views/OpenOrders.vue#L72